`name-replacements`: Exclude vite-env from rename by default

### Description

https://vite.dev/guide/env-and-mode#intellisense-for-typescript

Vite has some settings with env that I don't think works with "environment" instead of "env"

  1:1   error  The filename `vite-env.d.ts` should be named `vite-environment.d.ts`. A more descriptive name will do too  unicorn/name-replacements
  1:11  error  The variable `ImportMetaEnv` should be named `ImportMetaEnvironment`. A more descriptive name will do too  unicorn/name-replacements

These should not be errors I think

### Examples

interface ViteTypeOptions {
  // By adding this line, you can make the type of ImportMetaEnv strict
  // to disallow unknown keys.
  // strictImportMetaEnv: unknown
}

interface ImportMetaEnv {
  readonly VITE_APP_TITLE: string
  // more env variables...
}

interface ImportMeta {
  readonly env: ImportMetaEnv
}

All allowed 😎 👍 

### Additional Info

_No response_

--- a/rules/shared/name-replacements.js
+++ b/rules/shared/name-replacements.js
@@ -309,6 +309,9 @@ export const defaultAllowList = {
 	getInitialProps: true,
 	getServerSideProps: true,
 	getStaticProps: true,
+	// Vite environment variables type
+	// https://vite.dev/guide/env-and-mode#intellisense-for-typescript
+	ImportMetaEnv: true,
 	// The name iOS is a standard name for an OS
 	iOS: true,
 	// React PropTypes
@@ -327,4 +330,7 @@ export const defaultIgnore = [
 	'a11y', // Accessibility
 	'e2e', // End-to-end
 	'jQuery',
+	// Vite type declaration file
+	// https://vite.dev/guide/env-and-mode#intellisense-for-typescript
+	'vite-env',
 ];

--- a/test/name-replacements.js
+++ b/test/name-replacements.js
@@ -1988,7 +1988,10 @@ test.typescript({
 			},
 		},
 	},
-	valid: [],
+	valid: [
+		// Vite type declaration
+		'interface ImportMetaEnv {}',
+	],
 	invalid: [
 		// Types
 		...[
@@ -2470,6 +2473,11 @@ test({
 		{
 			code: 'foo();',
 			filename: 'e2e.js',
+		},
+		// Vite type declaration
+		{
+			code: 'foo();',
+			filename: 'vite-env.d.ts',
 		},
 		// `ignore` option
 		{

